### ISSUE TITLE
`LineStripsND` C++ constructors are unfriendly with multiple strips

### ISSUE THREAD

**Issue by teh-cmc** [rerun-io/rerun#3202]
I cannot seem to do this with our cpp API:
```c
std::vector<std::vector<rr::datatypes::Vec3D>> points = {
    {{0.f, 0.f, 0.f}, {0.f, 0.f, 1.f}},
    {{1.f, 0.f, 0.f}, {1.f, 0.f, 1.f}},
    {{1.f, 1.f, 0.f}, {1.f, 1.f, 1.f}},
    {{0.f, 1.f, 0.f}, {0.f, 1.f, 1.f}},
};
rec.log("segments", rr::LineStrips2D(points));
```

Instead I have to do this, which is quite annoying:
```c
std::vector<rr::datatypes::Vec3D> points1 = {{0.f, 0.f, 0.f}, {0.f, 0.f, 1.f}};
std::vector<rr::datatypes::Vec3D> points2 = {{1.f, 0.f, 0.f}, {1.f, 0.f, 1.f}};
std::vector<rr::datatypes::Vec3D> points3 = {{1.f, 1.f, 0.f}, {1.f, 1.f, 1.f}};
std::vector<rr::datatypes::Vec3D> points4 = {{0.f, 1.f, 0.f}, {0.f, 1.f, 1.f}};
rec.log("segments", rr::LineStrips3D({points1, points2, points3, points4}));
```

Not sure if I'm missing some C++ idiom or otherwise.

**Comment by emilk** [comment]
This is very related to #3330 

I think we should allow passing any container (something we can call `std::begin/end` on), including `std::vector` and initializer lists.

**Comment by Wumpf** [comment]
* Fixed by #3971
